### PR TITLE
Migrate to MongoDB 6

### LINKED ISSUE #1083: Migrate to MongoDB 6
We should update our tests to match MongoDB 6.0 behavior. See #1074.

This task is only about that; everything else is probably not in the scope. In particular, we should not:
* replace regular integration tests with compatibility tests or vice versa;
* change or refactor tests in any significant way;
* try to support both MongoDB 5 and 6;
* change our current decisions about error messages;
* etc.

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

Closes #1083.

## Readiness checklist

* [x] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [x] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by noisersup — APPROVED** [review]

**Comment by AlekSi** [comment]
@rumyantseva PTAL

File: integration/commands_diagnostic_test.go [hunk 1]
```diff
@@ -67,28 +67,35 @@ func TestCommandsDiagnosticGetLog(t *testing.T) {
 		"NonExistentName": {
 			command: bson.D{{"getLog", "nonExistentName"}},
 			err: &mongo.CommandError{
-				Code:    0,
-				Message: `no RamLog named: nonExistentName`,
+				Code:    96,
+				Name:    "OperationFailed",
+				Message: `No log named 'nonExistentName'`,
 			},
 			alt: `no RecentEntries named: nonExistentName`,
```
Comment (AlekSi) [inline integration/commands_diagnostic_test.go:74]: Do we need both messages there? I think they should be the same now

File: integration/commands_replication_test.go [hunk 2]
```diff
@@ -50,11 +50,11 @@ func TestCommandsReplicationIsMaster(t *testing.T) {
 			delete(m, "localTime")
 
 			maxWireVersion := m["maxWireVersion"].(int32)
-			assert.True(t, maxWireVersion == 0 || maxWireVersion == 13)
+			assert.True(t, maxWireVersion == 0 || maxWireVersion == 17)
```
Comment (AlekSi) [inline integration/commands_replication_test.go:53]: Let's add a comment explaining why we need `== 0` in those lines

File: integration/query_evaluation_test.go [hunk 3]
```diff
@@ -464,15 +482,25 @@ func TestQueryEvaluationRegex(t *testing.T) {
...
 		"RegexBadOption": {
-			filter:      bson.D{{"v", bson.D{{"$regex", primitive.Regex{Pattern: "foo", Options: "123"}}}}},
-			expectedIDs: []any{"multiline-string", "string"},
+			filter: bson.D{{"v", bson.D{{"$regex", primitive.Regex{Pattern: "foo", Options: "123"}}}}},
+			err: &mongo.CommandError{
+				Code:    51108,
+				Name:    "Location51108",
+				Message: ` invalid flag in regex options: 1`,
+			},
+			altMessage: `invalid flag in regex options: 1`,
```
Comment (AlekSi) [inline integration/query_evaluation_test.go:491]: I _think_ altMessage was supposed to be MongoDB's error, not vise versa.

Let's rename `AssertEqualAltError` argument names and update docs to mention that if that's true

**Review by rumyantseva — APPROVED** [review]
Yeah, that looks reasonable, and I don't see other places we might need to "migrate".

**Pull request merged**